sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -40,6 +40,12 @@ export interface ProxySettings {
   port: number;
   username?: string;
   password?: string;
+  key?: Buffer; // key for proxy server
+  ca?: Buffer[]; // ca for proxy server
+  cert?: Buffer; // cert for proxy server
+  headers?: { [key: string]: any }; // headers specific to proxy options
+  servername?: string;
+  localAddress?: string;
```

Comment:
In what scenario would we use `servername` or `localAddress` when we have the `host` property that the user can set?